pennylane/dla/lie_closure.py

```diff
@@ -143,16 +143,44 @@ def add(self, other):
         for ps in other:
             # TODO: Potential speed-up by computing the maximal linear independent set for all current basis vectors + other, essentially algorithm1 in https://arxiv.org/abs/1012.5256
             self._M, self._pw_to_idx, self._rank, self._num_pw, is_independent = (
-                self._add_if_independent(self._M, ps, self._pw_to_idx, self._rank, self._num_pw)
+                self._is_independent(self._M, ps, self._pw_to_idx, self._rank, self._num_pw)
             )
             if is_independent:
                 self._basis.append(ps)
         return self._basis
 
+    def is_independent(self, pauli_sentence, tol=1e-15):
+        r"""Check if the :class:`~PauliSentence` ``pauli_sentence`` is linearly independent with all vectors in ``PauliVSpace``.
+
+        This is done in the following way: ``M`` (see description in class) is extended by ``pauli_sentence``.
+        If the added operator has a PauliWord (key) that is new to ``pw_to_idx``, then we have to add a new row
+        and already know that it has to be linearly independent.
+        If it contains the same PauliWords, we need to compute the new rank and compare it with the old rank.
+        If the rank is the same, the operator is linearly dependent and not added. Else, the rank is incrased by 1
+        and the extended M becomes our new M.
```

Comment:
This description seems to imply that this function mutates `self._M`, which is not the case.

Reply:
✅